src/hotspot/os/windows/os_windows.cpp

```diff
@@ -3346,12 +3357,23 @@ void os::large_page_init() {
     return;
   }
 
-  _large_page_size = large_page_init_decide_size();
+  _large_page_size = os::win32::large_page_init_decide_size();
   const size_t default_page_size = os::vm_page_size();
   if (_large_page_size > default_page_size) {
+#if !defined(IA32)
+    if (EnableAllLargePageSizesForWindows) {
+      size_t min_size = GetLargePageMinimum();
+
+      // Populate _page_sizes with large page sizes less than or equal to _large_page_size, ensuring each page size is double the size of the previous one.
+      for (size_t page_size = min_size; page_size < _large_page_size; page_size *= 2) {
```

Comment:
Does Windows actually use these page sizes, or does it always use 2M pages?

Also, you're only adding powers of 2 here; the user-defined LargePageSizeInBytes doesn't need to be a power of two, and can be 6MB, for example. Will that be a problem?

Reply:
Windows can use different large page sizes based on hardware support, not limited to the default 2MB. The implementation examines the system's supported page sizes and selects an appropriate size using the LargePageSizeInBytes parameter. 
The code accommodates non-power-of-two sizes, such as 6MB, while ensuring these sizes are multiples of the system's minimum large page size to prevent potential issues. If an invalid size is specified, the code defaults to the minimum supported size and issues a warning.